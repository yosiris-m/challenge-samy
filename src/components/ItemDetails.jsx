import { useState } from "react";
import "./ItemDetails.scss";
import { sendLikes } from "../services/api";
import { LikedSvg } from "../icons/LikedSvg";
import { PriceTriangleSvg } from "../icons/PriceTriangleSvg";
import { RepostSvg } from "../icons/RepostSvg";

export function ItemDetails({ item }) {
  const [likesCount, setLikesCount] = useState(item.likesCount);
  const [isLiked, setIsLiked] = useState(item.isLiked);

  const addLike = () => {
    const newLikes = likesCount + 1;
    setLikesCount(newLikes);
    sendLikes(item.id);
    setIsLiked(true);
  };

  const removeLike = () => {
    const removeLike = likesCount - 1;
    setLikesCount(removeLike);
    setIsLiked(false);
  };

  return (
    <section className="item-details">
      <div className="item-details__image-container">
        <div className="item-details__price-bg">
          <PriceTriangleSvg />
        </div>
        <div className="item-details__price">
          <span>{item.price}</span>
          <span className="item-details__currency">â‚¬</span>
        </div>

        <img
          src={item.image}
          alt={item.title}
          className="item-details__image"
        />
      </div>

      <section className="item-info">
        <h1 className="item-info__title">{item.title}</h1>
        <p className="item-info__subtitle">
          <span className="item-info__by">by</span> {item.author}
        </p>
        <div className="item-actions">
          <div className="item-actions__like-container">
            <p className="item-actions__likes-count">
              {likesCount.toString().padStart(3, "0")}
            </p>
            {isLiked ? (
              <span onClick={removeLike} className="item-actions__like-img">
                <LikedSvg />
              </span>
            ) : (
              <span
                onClick={addLike}
                className="item-actions__like-img--selected"
              >
                <LikedSvg />
              </span>
            )}
          </div>
          <div className="item-actions__repost">
            <RepostSvg />
            <span className="item-actions__repost-counter">000</span>
          </div>
        </div>
      </section>
    </section>
  );
}
