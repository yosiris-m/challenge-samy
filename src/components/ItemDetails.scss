.item-details {
  margin-left: 2.5rem;
  margin-right: 2.5rem;
  margin-top: 1.25rem;
  font-family: "Brandon Grotesque";
  position: relative;
  border: 1px solid #e2e2e2;
  background-color: #fff;

  &__image-container {
    position: relative;
  }

  &__image {
    width: 100%;
    object-fit: cover;
  }

  &__price {
    position: absolute;
    top: 1.3125rem;
    left: 1.125rem;
    font-size: 1.25rem;
    line-height: 0.9375rem;
    text-transform: uppercase;
    color: #000000;
    font-weight: 500;
  }

  &__currency {
    font-size: 0.75rem;
    padding-left: 0.2rem;
  }

  &__price-bg {
    position: absolute;
    top: 0;
    left: -1px;
  }
}

.item-info {
  &__title {
    font-size: 1.8125rem;
    line-height: 1.5625rem;
    text-transform: uppercase;
    color: #000000;
    font-weight: 400;
    font-family: "Open Sans";
    text-align: center;
    margin: 0;
    padding-top: 1.5rem;
  }

  &__subtitle {
    font-size: 1.125rem;
    line-height: 1.5625rem;
    color: #000;
    font-weight: 400;
    font-family: "Droid Serif";
    text-align: center;
    margin: 0;
    margin-top: 0.625rem;
    padding-bottom: 1.5625rem;
    filter: drop-shadow(0.0313rem 0.0541rem 0px #ffffff);
  }

  &__by {
    color: #c2c2bd;
  }
}

.item-actions {
  display: flex;
  flex-direction: row;
  font-size: 23px;
  line-height: 10px;
  text-transform: uppercase;
  color: #b6b6b4;
  text-align: center;
  border-top: 1px solid #e2e2e2;
  height: 3.75rem;

  svg {
    fill: #b6b6b4;
    stroke: #b6b6b4;
  }

  @media all and (min-width: 768px) {
    position: absolute;
    display: block;
    top: 1.25rem;
    right: 1.1875rem;
    border-top: none;
    color: #fff;

    svg {
      fill: rgb(255, 255, 255);
      stroke: rgb(255, 255, 255);
    }
  }

  &__repost {
    display: flex;
    flex-grow: 1;
    flex-direction: row;
    align-items: center;
    border-left: 1px solid #e2e2e2;
    justify-content: center;

    @media all and (min-width: 768px) {
      border: none;
      flex-direction: column;
    }
  }

  &__repost-counter {
    margin-left: 1.125rem;

    @media all and (min-width: 768px) {
      margin-left: 0;
      margin-top: 1.125rem;
    }
  }

  &__like-container {
    display: flex;
    flex-grow: 1;
    flex-direction: row;
    align-items: center;
    justify-content: center;

    @media all and (min-width: 768px) {
      flex-direction: column-reverse;
    }
  }

  &__likes-count {
    @media all and (min-width: 768px) {
      margin-top: 0.5rem;
      margin-bottom: 1.125rem;
    }
  }

  &__like-img {
    margin-left: 1.2rem;
    cursor: pointer;

    @media all and (min-width: 768px) {
      margin-left: 0;
    }

    &--selected {
      margin-left: 1.2rem;

      svg {
        fill: #83f3c6;
        stroke: #83f3c6;
      }

      @media all and (min-width: 768px) {
        margin-left: 0;
      }
    }
  }
}
