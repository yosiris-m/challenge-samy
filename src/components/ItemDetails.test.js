import React from "react";

import { ItemDetails } from "./ItemDetails";
import renderer from "react-test-renderer";

it("should render ItemDetails when it is liked", () => {
  const item = {
    id: 4,
    title: "Red fruits",
    author: "SamyRoad",
    price: 10,
    image: "https://picsum.photos/id/102/300",
    liked: true,
    likesCount: 7,
  };
  const tree = renderer.create(<ItemDetails item={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render ItemDetails when it is not liked", () => {
  const item = {
    id: 4,
    title: "Red fruits",
    author: "SamyRoad",
    price: 10,
    image: "https://picsum.photos/id/102/300",
    liked: false,
    likesCount: 0,
  };
  const tree = renderer.create(<ItemDetails item={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});
