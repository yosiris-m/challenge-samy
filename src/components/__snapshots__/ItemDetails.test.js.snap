// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ItemDetails when it is liked 1`] = `
<div
  className="item-details"
>
  <div
    className="item-details__image-container"
  >
    <div
      className="item-details__price-bg"
    >
      <svg
        height="100"
        version="1.2"
        viewBox="0 0 157 99"
        width="160"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="item_ 4"
        >
          <g
            id="Price"
          >
            <path
              className="s0"
              d="m0-1h160l-160 100z"
              id="Rectangle 12"
              style={
                Object {
                  "fill": "#ffffff",
                }
              }
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      className="item-details__price"
    >
      <span>
        10
      </span>
      <span
        className="item-details__currency"
      >
        €
      </span>
    </div>
    <img
      alt="Red fruits"
      className="item-details__image"
      src="https://picsum.photos/id/102/300"
    />
  </div>
  <div
    className="item-info"
  >
    <p
      className="item-info__title"
    >
      Red fruits
    </p>
    <p
      className="item-info__subtitle"
    >
      <span
        className="item-info__by"
      >
        by
      </span>
       
      SamyRoad
    </p>
    <div
      className="item-actions"
    >
      <div
        className="item-actions__like-container"
      >
        <p
          className="item-actions__likes-count"
        >
          007
        </p>
        <div
          className="item-actions__like-img--selected"
          onClick={[Function]}
        >
          <svg
            height={40}
            width={40}
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <filter
                height="200%"
                id="f1"
                width="200%"
                x="-50%"
                y="-50%"
              >
                <feDropShadow
                  dx={0}
                  dy={0}
                  floodColor="#000"
                  floodOpacity={0.5}
                  stdDeviation={0.417}
                />
              </filter>
            </defs>
            <style />
            <g
              id="item_1"
            >
              <g
                id=":hover"
              >
                <g
                  id="likes"
                >
                  <g
                    id="liked"
                  >
                    <path
                      d="M20.5 3C30.2 3 38 10.8 38 20.5S30.2 38 20.5 38 3 30.2 3 20.5 10.8 3 20.5 3zm4 12.1c-.7-1.2-1.6-1.7-1.7-3v-1.6c-.2-.6-1.3-1.1-1.9-1.1-.7 0-1.5.6-1.5 1.3h-.1V15H13v.1c-.7 0-1 .7-1 1.4s.3 1.3 1 1.3h2.1v.9l-2.1-.1v.1c-.7 0-1 .7-1 1.4s.3 1.3 1 1.3h2.1v.7H13c-.7 0-1 .7-1 1.4s.3 1.4 1 1.4h2.1v.6h-2l-.1.2c-.7 0-1 .6-1 1.3 0 .7.3 1.3 1 1.4h12.7s.8-2.2 1.7-2.2h1.7v-8.4s-3.2-.4-4.6-2.7zm-1.7-4.6q.1.1.1.2c0-.1-.1-.3-.1-.2z"
                      fillRule="evenodd"
                      id="Ellipse 5 copy 5"
                      style={
                        Object {
                          "filter": "url(#f1)",
                          "strokeWidth": 0.5,
                        }
                      }
                    />
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        className="item-actions__repost"
      >
        <svg
          height="38"
          version="1.2"
          viewBox="0 0 38 38"
          width="38"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <filter
              height="200%"
              id="f1"
              width="200%"
              x="-50%"
              y="-50%"
            >
              <feDropShadow
                dx="0"
                dy="0"
                floodColor="#000000"
                floodOpacity=".3"
                stdDeviation="0.4166666666666667"
              />
            </filter>
          </defs>
          <g
            id="item_ 4"
          >
            <g
              id="repost"
            >
              <path
                className="s0"
                d="m19.4 1.5c9.6 0 17.5 7.9 17.5 17.5 0 9.7-7.9 17.5-17.5 17.5-9.7 0-17.5-7.8-17.5-17.5 0-9.6 7.8-17.5 17.5-17.5zm10.7 17.4c0-5.8-4.7-10.5-10.5-10.5-5.8 0-10.5 4.7-10.5 10.5h-2.1l3.5 4.2 3.5-4.2h-2.1c0-4.2 3.4-7.7 7.7-7.7 4.2 0 7.7 3.5 7.7 7.7 0 4.3-3.5 7.7-7.7 7.7v2.8c5.8 0 10.5-4.7 10.5-10.5z"
                fillRule="evenodd"
                id="Ellipse 5 copy 4 copia 2"
                style={
                  Object {
                    "strokeWidth": 0.4,
                  }
                }
              />
            </g>
          </g>
        </svg>
        <span
          className="item-actions__repost-counter"
        >
          000
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render ItemDetails when it is not liked 1`] = `
<div
  className="item-details"
>
  <div
    className="item-details__image-container"
  >
    <div
      className="item-details__price-bg"
    >
      <svg
        height="100"
        version="1.2"
        viewBox="0 0 157 99"
        width="160"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="item_ 4"
        >
          <g
            id="Price"
          >
            <path
              className="s0"
              d="m0-1h160l-160 100z"
              id="Rectangle 12"
              style={
                Object {
                  "fill": "#ffffff",
                }
              }
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      className="item-details__price"
    >
      <span>
        10
      </span>
      <span
        className="item-details__currency"
      >
        €
      </span>
    </div>
    <img
      alt="Red fruits"
      className="item-details__image"
      src="https://picsum.photos/id/102/300"
    />
  </div>
  <div
    className="item-info"
  >
    <p
      className="item-info__title"
    >
      Red fruits
    </p>
    <p
      className="item-info__subtitle"
    >
      <span
        className="item-info__by"
      >
        by
      </span>
       
      SamyRoad
    </p>
    <div
      className="item-actions"
    >
      <div
        className="item-actions__like-container"
      >
        <p
          className="item-actions__likes-count"
        >
          000
        </p>
        <div
          className="item-actions__like-img--selected"
          onClick={[Function]}
        >
          <svg
            height={40}
            width={40}
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <filter
                height="200%"
                id="f1"
                width="200%"
                x="-50%"
                y="-50%"
              >
                <feDropShadow
                  dx={0}
                  dy={0}
                  floodColor="#000"
                  floodOpacity={0.5}
                  stdDeviation={0.417}
                />
              </filter>
            </defs>
            <style />
            <g
              id="item_1"
            >
              <g
                id=":hover"
              >
                <g
                  id="likes"
                >
                  <g
                    id="liked"
                  >
                    <path
                      d="M20.5 3C30.2 3 38 10.8 38 20.5S30.2 38 20.5 38 3 30.2 3 20.5 10.8 3 20.5 3zm4 12.1c-.7-1.2-1.6-1.7-1.7-3v-1.6c-.2-.6-1.3-1.1-1.9-1.1-.7 0-1.5.6-1.5 1.3h-.1V15H13v.1c-.7 0-1 .7-1 1.4s.3 1.3 1 1.3h2.1v.9l-2.1-.1v.1c-.7 0-1 .7-1 1.4s.3 1.3 1 1.3h2.1v.7H13c-.7 0-1 .7-1 1.4s.3 1.4 1 1.4h2.1v.6h-2l-.1.2c-.7 0-1 .6-1 1.3 0 .7.3 1.3 1 1.4h12.7s.8-2.2 1.7-2.2h1.7v-8.4s-3.2-.4-4.6-2.7zm-1.7-4.6q.1.1.1.2c0-.1-.1-.3-.1-.2z"
                      fillRule="evenodd"
                      id="Ellipse 5 copy 5"
                      style={
                        Object {
                          "filter": "url(#f1)",
                          "strokeWidth": 0.5,
                        }
                      }
                    />
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        className="item-actions__repost"
      >
        <svg
          height="38"
          version="1.2"
          viewBox="0 0 38 38"
          width="38"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <filter
              height="200%"
              id="f1"
              width="200%"
              x="-50%"
              y="-50%"
            >
              <feDropShadow
                dx="0"
                dy="0"
                floodColor="#000000"
                floodOpacity=".3"
                stdDeviation="0.4166666666666667"
              />
            </filter>
          </defs>
          <g
            id="item_ 4"
          >
            <g
              id="repost"
            >
              <path
                className="s0"
                d="m19.4 1.5c9.6 0 17.5 7.9 17.5 17.5 0 9.7-7.9 17.5-17.5 17.5-9.7 0-17.5-7.8-17.5-17.5 0-9.6 7.8-17.5 17.5-17.5zm10.7 17.4c0-5.8-4.7-10.5-10.5-10.5-5.8 0-10.5 4.7-10.5 10.5h-2.1l3.5 4.2 3.5-4.2h-2.1c0-4.2 3.4-7.7 7.7-7.7 4.2 0 7.7 3.5 7.7 7.7 0 4.3-3.5 7.7-7.7 7.7v2.8c5.8 0 10.5-4.7 10.5-10.5z"
                fillRule="evenodd"
                id="Ellipse 5 copy 4 copia 2"
                style={
                  Object {
                    "strokeWidth": 0.4,
                  }
                }
              />
            </g>
          </g>
        </svg>
        <span
          className="item-actions__repost-counter"
        >
          000
        </span>
      </div>
    </div>
  </div>
</div>
`;
